src/signals/settings/users/containers/Overview/__tests__/Overview.test.js

```diff
@@ -521,74 +521,104 @@ describe('signals/settings/users/containers/Overview', () => {
 
-    // filters: user active = false
-    const userInactiveFilterValue = 'false';
-    act(() => {
-      fireEvent.change(filterByUserActiveSelect, { target: { value: userInactiveFilterValue } });
-    });
-    expect(dispatch).toHaveBeenCalledTimes(2);
+    expect(fetch).toHaveBeenCalledWith(
+      expect.stringContaining('is_active=true'),
+      expect.objectContaining(testContext.apiHeaders)
+    );
+  });
+
+  it('should select a value in the select filters and dispatch a fetch action', async () => {
+    jest
+      .spyOn(rolesSelectors, 'inputRolesSelector')
+      .mockImplementation(() => inputRolesSelectorJSON);
 
-    const filtersInactive = await fetch(`${apiPrefix}&is_active=false`);
-    const filtersInactiveResult = await filtersInactive.json();
-    expect(filtersInactiveResult.count).toBe(19);
+    const mockedState = { users: { filters: { is_active: true, role: 'Behandelaar' } } };
 
-    // filters: user active = true, role = Behandelaar
-    const filtersRoleValue = 'Behandelaar';
-    act(() => {
-      fireEvent.change(filterByRoleSelect, { target: { value: filtersRoleValue } });
-      fireEvent.change(filterByUserActiveSelect, { target: { value: userInactiveFilterValue } });
-    });
-    expect(dispatch).toHaveBeenCalledTimes(4);
+    const { getByTestId } = render(usersOverviewWithAppContext({}, {}, mockedState));
+
+    await wait();
 
-    const filtersActiveAndRole = await fetch(`${apiPrefix}&is_active=true&role=Behandelaar`);
-    const filtersActiveAndRoleResult = await filtersActiveAndRole.json();
-    expect(filtersActiveAndRoleResult.count).toBe(15);
+    const filterByRoleSelect = getByTestId('roleSelect');
+    const filterByUserActiveSelect = getByTestId('userActiveSelect');
+
+    expect(dispatch).toHaveBeenCalledTimes(0);
 
-    // filters: user active = false, role = Behandelaar
     act(() => {
-      fireEvent.change(filterByRoleSelect, { target: { value: filtersRoleValue } });
-      fireEvent.change(filterByUserActiveSelect, { target: { value: userInactiveFilterValue } });
+      fireEvent.change(filterByUserActiveSelect, { target: { value: true } });
+      fireEvent.change(filterByRoleSelect, { target: { value: 'Behandelaar' } });
     });
-    expect(dispatch).toHaveBeenCalledTimes(6);
 
-    const filtersInactiveAndRole = await fetch(`${apiPrefix}&is_active=false&role=Behandelaar`);
-    const filtersInactiveAndRoleResult = await filtersInactiveAndRole.json();
-    expect(filtersInactiveAndRoleResult.count).toBe(16);
+    expect(dispatch).toHaveBeenCalledTimes(2);
+
+    expect(filterByRoleSelect.value).toBe('Behandelaar');
+    expect(filterByUserActiveSelect.value).toBe('true');
+
+    expect(fetch).toHaveBeenCalledWith(
+      expect.stringContaining('role=Behandelaar'),
+      expect.objectContaining(testContext.apiHeaders)
+    );
+
+    expect(fetch).toHaveBeenCalledWith(
+      expect.stringContaining('is_active=true'),
+      expect.objectContaining(testContext.apiHeaders)
+    );
```

Comment:
I would combine these two into one assertion. Now it looks like there have been two fetch calls.
Also, assets the amount of fetch calls. Should've only been 1

Reply:
Done :)